### PR TITLE
Kinesis blog post

### PR DESCRIPTION
**Pull request by gunnarmorling**
@jpechane Feedback welcome :)

### CODE REVIEW

File: blog/2018-08-30-streaming-mysql-data-changes-into-kinesis.adoc [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+If you've ever set up the Debezium MySQL connector in Kafka Connect, most of the properties will look familiar to you.
+
+But let's talk about the `OFFSET_STORAGE` and `DATABASE_HISTORY` options in a bit more detail.
+They deal with how connector offsets and the database history should be persisted.
+When running the connector via Kafka Connect, both would typically be stored in specific Kafka topics.
+But that's not an option here, so an alternative is needed.
+For this example we're simply going to keep the offsets and database history in memory.
+I.e. if the engine is restarted, this information will be lost and the connector will start from scratch, e.g. with a new initial snapshot.
+
+While out of scope for this blog post, it wouldn't be too difficult to create alternative implementations of the `OffsetBackingStore` and `DatabaseHistory` contracts, respectively.
```
Comment (jpechane) [inline blog/2018-08-30-streaming-mysql-data-changes-into-kinesis.adoc]: I think file based stores can get honorable mention as they are available for limit ed produciton use cases

File: blog/2018-08-30-streaming-mysql-data-changes-into-kinesis.adoc [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+        .field("key", record.keySchema())
+        .field("value", record.valueSchema())
+        .build();
+
+    Struct message = new Struct(schema);
+    message.put("key", record.key());
+    message.put("value", record.value());
+
+    // create partition key by hashing the record's key
+    String partitionKey = String.valueOf(record.key() != null ? record.key().hashCode() : new Object().hashCode());
```
Comment (jpechane) [inline blog/2018-08-30-streaming-mysql-data-changes-into-kinesis.adoc]: Should not all records without key end up in the same partition?

File: blog/2018-08-30-streaming-mysql-data-changes-into-kinesis.adoc [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+
+    // create data blob representing the container by using Kafka Connect's JSON converter
+    final byte[] payload = valueConverter.fromConnectData("dummy", schema, message);
```
Comment (jpechane) [inline blog/2018-08-30-streaming-mysql-data-changes-into-kinesis.adoc]: I think this does not correspond to text - only value is sent, is not it?

**Comment by jpechane** [comment]
@gunnarmorling Very nice and easy to follow and understand! I've just added one note and otherwise we are good to go.

File: blog/2018-08-30-streaming-mysql-data-changes-into-kinesis.adoc (hunk 2 above)
Comment (gunnarmorling) [reply blog/2018-08-30-streaming-mysql-data-changes-into-kinesis.adoc]: Yes, good thinking. I'll set it to a fixed value.

File: blog/2018-08-30-streaming-mysql-data-changes-into-kinesis.adoc (hunk 1 above)
Comment (gunnarmorling) [reply blog/2018-08-30-streaming-mysql-data-changes-into-kinesis.adoc]: Ok, will do.

**Comment by gunnarmorling** [comment]
Addressed your remarks, applied a few more changes and merged. Thanks again for reviewing and your help, @jpechane! Btw. the broken rendering was caused apparently by an extra-whitespace after one of the post tags in the array within the document's pre-amble. Go figure...

**Pull request merged**